nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/tools/command/PositionCommand.java

```diff
@@ -0,0 +1,150 @@
+                .desc("Sets the current position to a record >= timestamp, eg. " + Instant.now()
+                + " The timestamp used depends on the implementation, for Kafka this is the LogAppendTime. "
+                + "If no record offset is found with an appropriate timestamp then the command fails.")
+                .hasArg()
+                .argName("TIMESTAMP")
+                .build());
+    }
+
+    @Override
+    public boolean run(LogManager manager, CommandLine cmd) throws InterruptedException {
+        String name = cmd.getOptionValue("log-name");
+        String group = cmd.getOptionValue("group", "tools");
+
+        if (cmd.hasOption("to-timestamp")) {
+            long timestamp = parseTimestamp(cmd.getOptionValue("to-timestamp"));
+            if (timestamp < 0) {
+                return false;
+            }
+            return positionToTimestamp(manager, group, name, timestamp);
+        } else if (cmd.hasOption("to-end")) {
+            toEnd(manager,group,name);
+        } else if (cmd.hasOption("reset")) {
+            reset(manager,group,name);
+        } else {
+            System.err.println("Invalid option, try 'help position'");
+            return false;
+        }
+
+        return true;
+    }
+
+    protected void toEnd(LogManager manager, String group, String name) {
+        LogLag lag = manager.getLag(name, group);
+
+        try (LogTailer<Record> tailer = manager.createTailer(group, name)) {
+            tailer.toEnd();
+        }
+        System.out.println(String.format("# Moved log %s, group: %s, from: %s to %s", name, group, lag.lag(), lag.upper()));
+    }
+
+    protected void reset(LogManager manager, String group, String name) {
+        LogLag lag = manager.getLag(name, group);
+        long pos = lag.lower();
+        try (LogTailer<Record> tailer = manager.createTailer(group, name)) {
+            tailer.reset();
+        }
+        System.out.println(String.format("# Reset log %s, group: %s, from: %s to 0", name, group, pos));
+    }
+
+    protected boolean positionToTimestamp(LogManager manager, String group, String name, long timestamp) {
+
+        try (LogTailer<Record> tailer = manager.createTailer(group, name)) {
+            int size = manager.getAppender(name).size();
+            for (int partition = 0; partition < size; partition++) {
+                LogOffset logOffset = tailer.offsetForTimestamp(new LogPartition(name, partition), timestamp);
+                if (logOffset != null) {
+                    tailer.seek(logOffset);
+                    tailer.commit();
+                    System.out.println(
+                            String.format("# Set log %s, group: %s, to offset %s", name, group, logOffset.offset()));
+                    return true;
+                } else {
+                    System.err.println(
+                            String.format("# Could not find an offset for log %s, group: %s", name, group));
+                }
+            }
+        }
+
```

Comment:
I think we should return false only if we were not able to commit any new positions. For instance it is not abnormal to have an empty partition or with timestamp that does not match, so return failure nothing where committed

Reply:
I believe that's what I had this morning but I changed it following your feedback `if the command fails (timestamp is not found or invalid) run method should return false (which will generate an error exit code)`.  To be clear, what should happen if the timestamp is not found?